imap_processing/codice/codice_l1a.py

```diff
@@ -84,6 +87,32 @@ def __init__(self, table_id: int, plan_id: int, plan_step: int, view_id: int):
         self.plan_step = plan_step
         self.view_id = view_id
 
+    def calculate_epoch_values(self) -> NDArray[int]:
+        """
+        Calculate and return the values to be used for `epoch`.
+
+        On CoDICE, the epoch values are derived from the `acq_start_seconds`
+        field in the packet.
+
+        Returns
+        -------
+        epoch : NDArray[int]
+            List of epoch values.
+        """
+        # Convert the acq_start_subseconds field from microseconds to seconds
+        subseconds = [value / 1000000 for value in self.dataset.acq_start_subseconds]
```

Comment:
I tried to see what type `self.dataset` is and couldn't find where `dataset` is set because it isn't in the `__init__` function. Are we guaranteed to have a `self.dataset` attribute?

The reason I was wondering about the type is because I think this function could be simplified to directly work on the arrays of values rather than going between lists and numpy functions.

```python
return met_to_j2000ns(self.data.acq_start_seconds + self.dataset.acq_start_subseconds / 1e6)